include/itkBoneMorphometryImageFilter.h

```diff
@@ -0,0 +1,175 @@
+ * its AfterThreadedGenerate method.
+ *
+ * \author: Jean-Baptiste Vimort
+ * \ingroup BoneMorphometry
+ *
+ */
+template< typename TInputImage>
+class ITK_TEMPLATE_EXPORT BoneMorphometryImageFilter:
+public ImageToImageFilter< TInputImage, TInputImage >
+{
+public:
+  /** Standard Self typedef */
+  typedef BoneMorphometryImageFilter                      Self;
+  typedef ImageToImageFilter< TInputImage, TInputImage >  Superclass;
+  typedef SmartPointer< Self >                            Pointer;
+  typedef SmartPointer< const Self >                      ConstPointer;
+
+  /** Method for creation through the object factory. */
+  itkNewMacro(Self);
+
+  /** Runtime information support. */
+  itkTypeMacro(BoneMorphometryImageFilter, ImageToImageFilter);
+
+  /** Image related typedefs. */
+  typedef typename TInputImage::Pointer    InputImagePointer;
+  typedef typename TInputImage::RegionType RegionType;
+  typedef typename TInputImage::SizeType   SizeType;
+  typedef typename TInputImage::IndexType  IndexType;
+  typedef typename TInputImage::PixelType  PixelType;
+
+  /** NeighborhoodIterator typedef */
+  typedef typename itk::ConstNeighborhoodIterator< TInputImage >    NeighborhoodIteratorType;
+  typedef typename NeighborhoodIteratorType::RadiusType             NeighborhoodRadiusType;
+  typedef typename NeighborhoodIteratorType::OffsetType             NeighborhoodOffsetType;
+
+  /** Type to use for computations. */
+  typedef typename NumericTraits< PixelType >::RealType RealType;
+
+  /** Methods to set/get the mask image */
+  itkSetInputMacro(MaskImage, TInputImage);
+  itkGetInputMacro(MaskImage, TInputImage);
+
+  /** Methods to set/get the mask image */
+  itkSetMacro(Threshold, RealType);
+  itkGetMacro(Threshold, RealType);
+
+  /** Methods to get the mask different outputs */
+
+  double GetBVTV() { return m_Pp; }
+
+  double GetxTbN() { return m_PlX; }
+  double GetyTbN() { return m_PlY; }
+  double GetzTbN() { return m_PlZ; }
+  double GetTbN() { return m_Pl; }
+
+  double GetxTbTh() { return m_Pp/m_PlX; }
+  double GetyTbTh() { return m_Pp/m_PlY; }
+  double GetzTbTh() { return m_Pp/m_PlZ; }
+  double GetTbTh() { return m_Pp/m_Pl; }
+
+  double GetxTbSp() { return (1.0 - m_Pp) / m_PlX; }
+  double GetyTbSp() { return (1.0 - m_Pp) / m_PlY; }
+  double GetzTbSp() { return (1.0 - m_Pp) / m_PlZ; }
+  double GetTbSp() { return (1.0 - m_Pp) / m_Pl; }
+
+  double GetxBSBV() { return 2.0 * (m_PlX / m_Pp); }
+  double GetyBSBV() { return 2.0 * (m_PlY / m_Pp); }
+  double GetzBSBV() { return 2.0 * (m_PlZ / m_Pp); }
+  double GetBSBV() { return 2.0 * (m_Pl / m_Pp); }
+
+
+protected:
+  BoneMorphometryImageFilter();
+  virtual ~BoneMorphometryImageFilter() {}
+
+
+  /** Pass the input through unmodified. Do this by Grafting in the
+  *  AllocateOutputs method.
+  */
+  void AllocateOutputs() ITK_OVERRIDE;
+
+  /** Override since the filter needs all the data for the algorithm */
```

Comment:
Probably, is EnlargeOutputRequestedRegion preventing to make the algorithm streamable?